lib/chef/application/client.rb

```diff
@@ -280,7 +287,14 @@ def reconfigure
       FileUtils.mkdir_p(Chef::Config.chef_repo_path)
       tarball_path = File.join(Chef::Config.chef_repo_path, 'recipes.tgz')
       fetch_recipe_tarball(Chef::Config[:recipe_url], tarball_path)
-      result = shell_out!("tar zxvf #{tarball_path} -C #{Chef::Config.chef_repo_path}")
+      if Chef::Config[:github]
+        cookbooks_dir = File.join(Chef::Config.chef_repo_path, 'cookbooks')
+        FileUtils.rm_rf(cookbooks_dir, :secure => true)
```

Comment:
Can we do something other than `rm_rf`?!  that feels wrong and likely to cause someone to have a really bad day.